source/code/plugin/KubernetesApiClient.rb

```diff
@@ -210,6 +211,123 @@ def getContainerLogsSinceTime(namespace, pod, container, since, showTimeStamp)
                 end
                 return containerLogs
             end
+
+            def getContainerResourceRequestsAndLimits(metricJSON, metricCategory, hostName, metricNameToCollect, metricNametoReturn)
+                metricItems = []
+                begin
+                    metricInfo = metricJSON
+                    metricInfo['items'].each do |pod|
+                        podUid = pod['metadata']['uid']
+                        if (!pod['spec']['containers'].nil?)
+                            pod['spec']['containers'].each do |container|
+                                containerName = container['name']
+                                if (!container['resources'].nil? && !container['resources'][metricCategory].nil? && !container['resources'][metricCategory][metricNameToCollect].nil?)
+                                    metricValue = getMetricNumericValue(metricNameToCollect, container['resources'][metricCategory][metricNameToCollect])
+                                    metricTime = Time.now.utc.iso8601 #2018-01-30T19:36:14Z
+
+                                    metricItem = {}
+                                    metricItem['DataItems'] = []
+                                    
+                                    metricProps = {}
+                                    metricProps['Timestamp'] = metricTime
+                                    metricProps['Host'] = hostName
+                                    metricProps['ObjectName'] = "K8SContainer"
+                                    metricProps['InstanceName'] = podUid + "/" + containerName
+                                    
+                                    metricProps['Collections'] = []
+                                    metricCollections = {}
+                                    metricCollections['CounterName'] = metricNametoReturn
+                                    metricCollections['Value'] = metricValue
+
+                                    metricProps['Collections'].push(metricCollections)
+                                    metricItem['DataItems'].push(metricProps)
+                                    metricItems.push(metricItem)
+                                end
+                            end
+                        end
+                    end
+                    rescue => error
+                    @Log.warn("getcontainerResourceRequestsAndLimits failed: #{error} for metric #{metricCategory} #{metricNameToCollect}")
+                    return metricItems
+                end
+                return metricItems          
+            end #getContainerResourceRequestAndLimits
+
+            def getMetricNumericValue(metricName, metricVal)
```

Comment:
Same as above, consider changing get* to parse*

Reply:
Also consider moving both getContainerResourceRequestsAndLimits and getMetricNumericValue to in_kube_podinventory  itself. KubernetesApiClient.rb is written as a client to talk to the kube api and deal with properties that are applicable to the cluster as a whole e.g. ClusterName .